### DISCUSSION TITLE
Any plans on releasing light-weight version?

### DISCUSSION THREAD

**Discussion by roycho111** [realm/realm-dotnet#3604] [Q&A] [answered]
Hello

I'm using Realm dotnet on Unity and i only use local database feature.

For a user that only uses local database feature, one might find that the SDK's file size is quite big.

As for Android arm64, librealm-wrappers.so file size is about 13mb.

It would be cool if there was a light-weight version only for local database users.

**Comment by nirinchev** [comment, marked as answer]
No, we don't plan to release a local-only version of Realm. When we last compared it, the sync API only added 1-2 MB compared to the local-only API, which didn't justify the increased complexity of managing multiple packages and ensuring we have a clear separation of the API. We understand that it'd benefit a portion of our userbase, but this is a game of diminishing returns (e.g. if we did split the packages, there'd be some people who don't need Decimal128 support or notifications), who'd benefit from further breaking down the package.

The good news is that Realm is fully open source, so if the size is a major concern, you could build the native libraries yourself - you'd need to pass in [`REALM_ENABLE_SYNC = OFF`](https://github.com/realm/realm-core/blob/master/CMakeLists.txt#L257) in your Cmake invocation to build the native library without sync support. We don't respect that config option in the .NET C API, so you'd also need to remove some code in [these files](https://github.com/realm/realm-dotnet/tree/main/wrappers/src). It's not exactly trivial, but should very much be doable in a couple of days.

**Comment by charlesroddie** [comment]
20.0.0 is local-only
